lib/Service/RoomService.php

```diff
@@ -167,7 +171,20 @@ public function createConversation(int $type, string $name, ?IUser $owner = null
 			throw new InvalidArgumentException('object');
 		}
 
-		$room = $this->manager->createRoom($type, $name, $objectType, $objectId);
+		if ($type !== Room::TYPE_PUBLIC || !$this->config->isPasswordEnforced()) {
+			$room = $this->manager->createRoom($type, $name, $objectType, $objectId);
+		} elseif ($password === '') {
+			throw new PasswordException(PasswordException::REASON_VALUE, $this->l10n->t('Password needs to be set'));
+		} else {
+			$event = new ValidatePasswordPolicyEvent($password);
+			try {
+				$this->dispatcher->dispatchTyped($event);
+			} catch (HintException $e) {
+				throw new PasswordException(PasswordException::REASON_VALUE, $e->getHint());
+			}
+
+			$room = $this->manager->createRoom($type, $name, $objectType, $objectId, $password);
+		}
```

Comment:
> This code must be created by the system for each room.

Does this mean it should be generated automatically in backend and exposed to user after room creation? So user does not need to add password when creating the room.

Reply:
No, the frontend must send it. But the code is all there already (apart from checking the enforcement in the web UI), it just sends a follow up request at the moment, and in the future it needs to be part of the creation request (if the capability is there and on)